lib/board.rb

```diff
@@ -0,0 +1,26 @@
+class Board
+  attr_accessor :board
+  attr_reader :win_patterns
+
+  def initialize
+    @board = [0, 1, 2, 3, 4, 5, 6, 7, 8]
+    @win_patterns = [ [0, 1, 2],[3, 4, 5],[6, 7,8],[0, 3, 6],[1, 4, 7],[2, 5, 8],[0, 4, 8],[2, 4, 6] ]
+  end
+
+  def add_at(pos, symbol)
+    @board[pos] = symbol
+  end
+
+  def full?
+    @board.each do |value|
+      return false if value.is_a? Integer
+    end
+    true
+  end
+
+  def taken?(value)
```

Comment:
There's a shorter and more proper way to write this method:

`@board[value].is_a? Integer`

That's it, it will return either `true` or `false` already.

Better yet you don't need this method anymore here in the `Board` as it's just a one-liner anyway. Please see my comment on `interface.rb`